### ISSUE TITLE
Pygame 2.0.2.dev2 -> NOFRAME and RESIZABLE flags not working when switching from fullscreen back to windowed mode

### ISSUE THREAD

**Issue by tank-king** [pygame/pygame#2711]
**Environment:**

- **Operating system** Windows 10
- **Python version** Python 3.9.5
- **SDL version** SDL 2.0.16
- **PyGame version** Pygame 2.0.2.dev2

**Current behavior:**

The NOFRAME and RESIZABLE flags do not work when switching from fullscreen mode back to windowed mode, and a non-resizable 'framed' window is set, but the SCALED flag works fine for all cases. This issue can be overriden when the window is set at least 2 times after reverting from fullscreen mode. 

**Expected behavior:**

When window is switched to windowed mode from fullscreen to windowed mode with the NOFRAME or RESIZABLE flags (or both of them), then the window should have no title bar, and be resizable respectively.

**Screenshots**

Before fullscreen toggle:
![image](https://user-images.githubusercontent.com/72032294/132944763-301cb023-8a53-4b1b-aae5-d6dddea39999.png)

In fullscreen mode:
![image](https://user-images.githubusercontent.com/72032294/132944800-3fbcdb5c-1eda-4d1e-bc18-35d082c3ffd8.png)

Switching to windowed mode after fullscreen mode with NOFRAME flag:
![image](https://user-images.githubusercontent.com/72032294/132944824-ae6ea5a9-1c9f-4ffd-8b0b-ec907a12b410.png)

**Steps to reproduce:**

1. open command prompt or powershell
2. pip install pygame==2.0.2.dev2
3. press any key twice after running the example code

**Test code**

```
import pygame

pygame.init()

fullscreen = False
size = (1280, 720)

full_flags = pygame.FULLSCREEN | pygame.SCALED
normal_flags = pygame.NOFRAME | pygame.RESIZABLE | pygame.SCALED


def toggle_fullscreen(f):
    if f:
        return pygame.display.set_mode(size, full_flags)
    # pygame.display.set_mode(size, normal_flags)  # uncomment this to see issue being fixed as a workaround
    return pygame.display.set_mode(size, normal_flags)


screen = pygame.display.set_mode(size, normal_flags)

clock = pygame.time.Clock()

while True:
    screen.fill("purple")

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            pygame.quit()
            raise SystemExit

        if event.type == pygame.KEYDOWN:
            fullscreen = not fullscreen
            screen = toggle_fullscreen(fullscreen)

    pygame.display.flip()
    clock.tick(144)  # framerate

```

**Comment by Starbuck5** [comment]
Additional context from me-

tank-king and I discussed this in DM on discord before posting the issue

- This issue is only apparent after #2460, but I doubt it's a regression from there
- I think this is low priority
- This has a high likelihood of being platform dependent
